### DISCUSSION TITLE
Static vs Dynamic

### DISCUSSION THREAD

**Discussion by nashysolutions** [pointfreeco/swift-composable-architecture#2305] [Q&A]
The TCA architecture uses a pure SPM approach for dependency management. All of its downstream dependencies are not explicitly declared as dynamic, so I assume they are statically linked. 

Does this not result in a larger binary, given that potentially multiple copies of linked packages are included?

**Comment by mbrandonw** [comment]
No I do not think TCA's dependencies are duplicated and does not result in a larger binary. As an example, the App Clip for [isowords](http://github.com/pointfreeco/isowords) must have a binary of less than 10mb yet it consists of many modules that each depend on TCA. 

Do you have a concrete example of where you are seeing the compiled binary become larger due to TCA?

**Reply by nashysolutions** [reply]
No, I don't suspect TCA is doing anything wrong. And I haven't seen any issues. I just wanted to reach out to someone who might help me understand. Stack-overflow, the swift forums and Apples documentation don't make this subject very clear,

https://forums.swift.org/t/include-dynamic-library-from-repo-i-dont-own-on-github/66219
